### PR TITLE
Prevents unhandledrejection error when checking for latest Astro version 

### LINKED ISSUE #11012: Check latest astro version fail cause astro render error page at first time visit.
### Astro Info

```block
Astro                    v4.8.2
Node                     v18.18.1
System                   Windows (x64)
Package Manager          pnpm
Output                   static
Adapter                  none
Integrations             none
```

### If this issue only occurs in one browser, which browser is a problem?

_No response_

### Describe the Bug

Astro cannot catch [fetchLatestAstroVersion](https://github.com/withastro/astro/blob/main/packages/astro/src/core/dev/dev.ts#L51) error when fetch failed, and it will create a error layout when first time fetch the page.

This bug may only appear in some country that cannot connect to npmjs.org normally.
It cannot reproduction in the stackblitz. I only can put some screenshots to show it.

The error will cause my test fail when I fetch the page first time.

Screenshots:
图片
图片

Stackblitz reproduction can follow this:
- Modify `node_modules/astro/dist/cli/install-package.js` and let `fetchPackageVersions` always return Error.
- Modify `node_modules/astro/dist/core/dev/dev.js#L31` and made shouldCheck to `true`.

图片

### What's the expected result?

Expect cli do nothing, keep server run normally.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-82f59z

### Participation

- [X] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by QingXia-Ela**
## Changes

Fix #11012

## Testing

Running test locally.

## Docs

It doesn't need to update docs.

### CODE REVIEW

File: packages/astro/src/core/dev/dev.ts [hunk 1]
```diff
@@ -48,29 +48,33 @@ export default async function dev(inlineConfig: AstroInlineConfig): Promise<DevS
...
+								await msg.newVersionAvailable({
+									latestVersion: version,
+								})
+							);
+						}
+					} catch (e) {
+						// Do nothing, check version fail shouldn't block server running.
 					}
 				}
 			});
```
Comment (bluwy) [inline packages/astro/src/core/dev/dev.ts:80]: I think the fix is to append a `.catch(() => {})` here instead. The outer `try..catch` intent was to ignore the error, but it doesn't work when the code uses `.then()`.

File: .changeset/warm-schools-applaud.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"astro": patch
+---
+
+fix: prevent error appear when check latest version fail(#11012)
```
Comment (bluwy) [inline .changeset/warm-schools-applaud.md:5]: Updated this to match the usual formatting.
```suggestion
Prevents unhandledrejection error when checking for latest Astro version
```

File: packages/astro/src/core/dev/dev.ts (hunk 1 above)
Comment (QingXia-Ela) [reply packages/astro/src/core/dev/dev.ts]: Done in 3c53f9593ab87a82afc913a20dea274212c3c3ac.

**Review by ematipico — APPROVED** [review]

**Review by florian-lefebvre — APPROVED** [review]

**Comment by matthewp** [comment]
Thank you!

**Pull request merged**